python/dgl/distributed/kvstore.py

```diff
@@ -362,6 +365,9 @@ def process_request(self, server_state):
         meta = {}
         kv_store = server_state.kv_store
         for name, data in kv_store.data_store.items():
+            if server_state.keep_alive:
+                if name not in kv_store.orig_data:
+                    continue
```

Comment:
Not sure why you need to keep a separate `orig_data` names.

Reply:
The reason is to map the data that exist when DistGraphServer is constructed only. This check aims to avoid read/write race condition if multiple client groups are booted simultaneously. some new DistTensor may be added by Group_A, but these tensors may be not ready yet when Clients in Group_B is initializing DistGraph. So only the totally ready Tensor is mapped to Group_B.

Reply:
yes. I am working on it and implement such logic in `launch.py` so that users could pay minimum attention on these constraints.